include/emp/bits/BitVector.hpp

```diff
@@ -701,6 +700,72 @@ namespace emp {
       return (int) (pos_found + (field_id * FIELD_BITS));
     }
 
+    /// Pop the last bit in the vector
+    bool PopBack() {
+      bool val = Get(num_bits-1);
+      if (val)  Set(num_bits-1, 0);
+      Resize(num_bits - 1);
+      return val;
+    }
+
+    /// Push given bit(s) onto the vector
+    void PushBack(bool bit=true, size_t num=1) {
+      Resize(num_bits + num);
+      for (size_t i=0; i < num; i++)
+        Set(num_bits-1, bit);
+    }
+
+    // Returns mask that keeps bits higher than index
+    void Mask_High(size_t index) {
+      size_t field_id = FieldID(index);
+      int field_t_bits = sizeof(bit_set[0]) * 8;
+      for (size_t i = 0; i < field_id; i++)
+        bit_set[i] = 0;
+      if (index >= 0)
+        bit_set[field_id] &= (MaskLow<field_t>((num_bits - index - 1) % field_t_bits) << (index + 1));
+    }
+
+    // Returns mask that keeps bits lower than index
+    void Mask_Low(size_t index) {
+      size_t field_id = FieldID(index);
+      int field_t_bits = sizeof(bit_set[0]) * 8;
+      for (size_t i = field_id + 1; i < NumFields(); i++)
+        bit_set[i] = 0;
+      if (index < num_bits)
+        bit_set[field_id] &= MaskLow<field_t>(index % field_t_bits);
+    }
+
+    /// Insert bits into vector, including middle using bitmagic
+    /// https://devolab.org/?p=2249
+    void Insert(size_t index, bool value=true, size_t num=1) {
+      Resize(num_bits + num);
+      // mask high and shift left: 101101 -> 1010000
+      thread_local BitVector a{};
+      a = *this;
+      if (index > 0)
+        a.Mask_High(index + num - 2);
+      a <<= num;
+      // mask low and | together 101101 -> 000101 ->1010101
+      thread_local BitVector b{};
+      b = *this;
+      b.Mask_Low(index);
+      a |= b;
+      // set specified bits -> 1010101 -> 101_101
+      for (size_t i = index; i < index + num; i++) {
+        a.Set(i, value);
+      }
+      std::swap(a, *this);
+
+    }
+
+    /// Delete bits from vector, including middle
+    void Delete(size_t index, size_t num=1) {
+      for (size_t j = index; j < num_bits - num; j++) {
```

Comment:
where possible, try to name iterator variables descriptively (for example, bit in this case) or default to i instead of j